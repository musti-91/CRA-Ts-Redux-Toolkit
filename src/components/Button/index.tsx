import React from 'react';

export interface Props {
  label: string;
  onPressed: () => void;
}

/**
 * @author
 * @function Button
 **/

const Button: React.FC<Props> = ({ label, onPressed }) => {
  return (
    <div>
      <button onClick={() => onPressed()}>{label}</button>
    </div>
  );
};

export default Button;
