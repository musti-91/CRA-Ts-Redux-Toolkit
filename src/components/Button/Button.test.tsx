import React from 'react';
import { shallow } from 'enzyme';
import Button from './index';

const props = {
  title: 'my-button',
  onPressed: jest.fn(),
};
describe('Button Component', () => {
  it('should match its snapshot', () => {
    const wrapper = shallow(<Button {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  // trigger
  it('should trigger onPressed', () => {
    const wrapper = shallow(<Button {...props} />);

    wrapper.find('button').prop('onClick')();

    expect(props.onPressed).toHaveBeenCalled();
  });
});
