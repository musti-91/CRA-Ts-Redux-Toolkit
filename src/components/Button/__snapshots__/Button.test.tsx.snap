// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match its snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
  />
</div>
`;
